### PR TITLE
feat:adds base commands for blueprint metadata

### PR DESCRIPTION
**Pull request by g-awmalik**
This adds a top level `blueprint` command to  the `cft` cli and adds the following commands as sub-commands to it:
- `test`
- `builds`
- `metadata` - new and in-progress

`cft test` will continue to work until we change that to `cft blueprint test` and take `test` out of the root command.

### CODE REVIEW

**Comment by g-awmalik** [comment]
@bharathkkb - can you take a look at this?

File: cli/cmd/blueprint.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	blueprintCmd.AddCommand(bpbuild.Cmd)
+	blueprintCmd.AddCommand(bptest.Cmd)
+
+	rootCmd.AddCommand(blueprintCmd)
+}
+
+var blueprintCmd = &cobra.Command{
+	Use:   "blueprint",
+	Short: "Blueprint CLI",
+	Long:  `The CFT blueprint CLI is used to execute commands specific to blueprints such as test, avgtime & metadata`,
```
Comment (bharathkkb) [inline cli/cmd/blueprint.go:21]:
```suggestion
	Long:  `The CFT blueprint CLI is used to execute commands specific to blueprints such as test, builds & metadata`,
```

File: cli/bpmetadata/cmd.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+func init() {
+	viper.AutomaticEnv()
+
+	MdCmd.Flags().StringVar(&mdFlags.path, "path", ".", "Path to the blueprint for generating metadata.")
+	MdCmd.Flags().BoolVar(&mdFlags.nested, "nested", false, "Flag for generating metadata for nested blueprint, if any.")
+}
+
+var MdCmd = &cobra.Command{
```
Comment (bharathkkb) [inline cli/bpmetadata/cmd.go:21]: Since this will be `bpmetadata.cmd`.
```suggestion
var cmd = &cobra.Command{
```

File: cli/bpmetadata/main.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package bpmetadata
+
+import (
+	log "github.com/inconshreveable/log15"
+)
+
+// bpbuild log15 handler
```
Comment (bharathkkb) [inline cli/bpmetadata/main.go:7]:
```suggestion
// bpmetadata log15 handler
```

**Review by bharathkkb — APPROVED** [review]

File: cli/bpmetadata/cmd.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	path   string
+	nested bool
+	verfiy bool
+}
+
+func init() {
+	viper.AutomaticEnv()
+
+	MdCmd.Flags().StringVar(&mdFlags.path, "path", ".", "Path to the blueprint for generating metadata.")
+	MdCmd.Flags().BoolVar(&mdFlags.nested, "nested", false, "Flag for generating metadata for nested blueprint, if any.")
```
[the remark this answers is not in the document]
Comment (g-awmalik) [reply cli/bpmetadata/cmd.go:18]: ack

File: cli/bpmetadata/cmd.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var mdFlags struct {
+	path   string
+	nested bool
+	verfiy bool
```
[the remark this answers is not in the document]
Comment (g-awmalik) [reply cli/bpmetadata/cmd.go:11]: Yeah forgot to take it out. Gone now.

File: cli/bpmetadata/cmd.go (hunk 2 above)
Comment (g-awmalik) [reply cli/bpmetadata/cmd.go:21]: You're right but it'll be `Cmd`. Changed it.

**Pull request merged**